src/query/sql/src/planner/binder/select.rs

```diff
@@ -295,6 +296,10 @@ impl Binder {
 
         s_expr = self.bind_projection(&mut from_context, &projections, &scalar_items, s_expr)?;
 
+        // rewrite async function to async function plan
+        let mut async_func_rewriter = AsyncFunctionRewriter::new();
+        s_expr = async_func_rewriter.rewrite(&s_expr)?;
+
```

Comment:
Is `AsyncFunctionCall` allowed to execute on where, having, join, etc.? If not, it needs to be checked in the binder phase and return a semantic error to avoid an unpredictable panic.